packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -1991,6 +2006,83 @@ describe('TransactionController', () => {
 
       expect(failedTx?.replacedBy).toBe(externalTransactionHash);
     });
+
+    it('updates post transaction balance if type is swap', async () => {
+      jest.useFakeTimers();
+
+      const mockPostTransactionBalanceUpdatedListener = jest.fn();
+      const controller = newController();
+      controller.hub.on(
+        TransactionEvent.postTransactionBalanceUpdated,
+        mockPostTransactionBalanceUpdatedListener,
+      );
+
+      const mockPostTxBalance = '7a00';
+      const externalTransactionToConfirm = {
+        from: ACCOUNT_MOCK,
+        to: ACCOUNT_2_MOCK,
+        id: '1',
+        chainId: '0x1',
+        status: TransactionStatus.confirmed,
+        type: TransactionType.swap,
+        txParams: {
+          gasUsed: undefined,
+          from: ACCOUNT_MOCK,
+          to: ACCOUNT_2_MOCK,
+        },
+        preTxBalance: '8b11',
+        // Default token address
+        destinationTokenAddress: '0x0000000000000000000000000000000000000000',
+      } as any;
+      const externalTransactionReceipt = {
+        gasUsed: '0x5208',
+      };
+      const externalBaseFeePerGas = '0x14';
+      const mockApprovalTransactionMeta = {
+        id: '2',
+      };
+
+      updatePostTransactionBalanceMock.mockImplementationOnce(
+        async (transactionMeta: TransactionMeta, _request: any) => {
+          return Promise.resolve({
+            updatedTransactionMeta: {
+              ...transactionMeta,
+              postTxBalance: mockPostTxBalance,
+            },
+            approvalTransactionMeta:
+              mockApprovalTransactionMeta as TransactionMeta,
+          });
+        },
+      );
+
+      await controller.confirmExternalTransaction(
+        externalTransactionToConfirm,
+        externalTransactionReceipt,
+        externalBaseFeePerGas,
+      );
+
+      jest.runAllTimers();
+
+      // This timeout will not prevent test to be wait for assertion since we are using fake timers, letting next tick to be executed since we are not awaiting for updatePostTransactionBalanceMock
+      setTimeout(() => {
```

Comment:
As we're mocking the `updatePostTransactionBalance` method, I'm not sure I understand why we need to use fake timers and `setTimeout` in this unit test?

I appreciate we're not awaiting our new method, but could we just use a `flushPromises` method to do that?

I think the typical implementation is:

```
await new Promise(jest.requireActual('timers').setImmediate);
```

Reply:
That was a left over from the old approach, removed that.